bindings/java/src/main/java/org/apache/opendal/exception/ODException.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.opendal.exception;
+
+public class ODException extends RuntimeException {
+    private final Code code;
+
+    public ODException(byte code, String message) {
+        this(Code.getCode(code), message);
+    }
+
+    public ODException(Code code, String message) {
+        super(message);
+        this.code = code;
+    }
+
+    public Code getCode() {
+        return code;
+    }
+
+    public enum Code {
+        Unexpected,
+        Unsupported,
+        ConfigInvalid,
+        NotFound,
+        PermissionDenied,
+        IsADirectory,
+        NotADirectory,
+        AlreadyExists,
+        RateLimited,
+        IsSameFile,
+        ConditionNotMatch,
+        ContentTruncated,
+        ContentIncomplete,
+        ;
+
+        public static Code getCode(byte value) {
+            for (final Code code : Code.values()) {
+                if (code.ordinal() == value){
+                    return code;
+                }
+            }
+            throw new IllegalArgumentException(
```

Comment:
I suggest returning a default code instead of throwing IllegalArgumentException.

Reply:
What is the "default code"? cc @Xuanwo

Reply:
Maybe return `UnkownError`. IllegalArgumentException usually means the user input param is wrong.

Reply:
Users are not expected to use this method with arbitrary input, we already add a test for enum exhausted and match https://github.com/apache/incubator-opendal/pull/2276/commits/c216fab616da669d93c95649fadc5bd88dba3402. Thus, this is an error/exception that something "Unknown" but legal.